server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -0,0 +1,97 @@
+package services.ti;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import forms.UpdateApplicantDob;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import models.Account;
+import models.Applicant;
+import models.TrustedIntermediaryGroup;
+import play.data.Form;
+import repository.SearchParameters;
+import repository.UserRepository;
+import services.DateConverter;
+import services.applicant.exception.ApplicantNotFoundException;
+
+public class TrustedIntermediaryService {
+
+  private final DateConverter dateConverter;
+  private final UserRepository userRepository;
+
+  @Inject
+  public TrustedIntermediaryService(DateConverter dateConverter, UserRepository userRepository) {
+    this.dateConverter = Preconditions.checkNotNull(dateConverter);
+    this.userRepository = Preconditions.checkNotNull(userRepository);
+  }
+
+  public ImmutableList<Account> retriveSearchResult(
```

Comment:
Suggestion, remove this method since it essentially has the same method signature as the "filter" method below.

Also note that calling trustedIntermediaryGroup.get() will throw if the optional is empty, so we need to handle that error (likely in the controller).